### DISCUSSION TITLE
Memory Leak in Celery Beat on Python 3.11.4

### DISCUSSION THREAD

**Discussion by esauser** [celery/celery#8388] [Q&A]
I went to submit an issue and it says I have to start a discussion first, so doing the needful.  We made no changes to our application other than changing the base image from `python-3.11.1-slim` to `python-3.11.4-slim`.  In doing so, Celery Beat ends up consuming seemingly infinite memory.  The only action that seems to reduce it is to restart the pods.  We did not have this issue prior to changing the base image, and the issue goes away when we change it back.

Interestingly, it doesn't seem to be consumed by the working process.  We can see that the process sits idle just like it did, but for whatever reason Kubernetes continues to allocate ever more memory, at a rate of about 8MB per hour.

There are absolutely other discussions about memory, but I didn't see any specifically related to changing the python patch version.

**Comment by auvipy** [comment]
can you use any memory leak identification tools to find out the root of it and share with us?

**Reply by esauser** [reply]
Do you have an example of one? I'm not familiar with this kind of troubleshooting.

**Reply by auvipy** [reply]
https://pypi.org/project/memory-profiler/ and https://pypi.org/project/memray/ are some example tool

**Reply by esauser** [reply]
We only see the issue in our deployed environment, slowly building over a relatively long period of time (few days).  We do not have the bandwidth to revert back to python 3.11.4 and attempt to utilize these tools in our deployed environment when we have to remain on 3.11.1 for our actual environments.
